### PR TITLE
mockstore: update pd/client and client-go to repair kvproto compatibility

### LINKED ISSUE #40598: repair kvproto compatibility
## Enhancement

Because https://github.com/pingcap/kvproto/pull/1035 will introduce some compatibility problems, we need to repair kvproto compatibility.

### PR DESCRIPTION
**Pull request by HuSharp**
Signed-off-by: husharp <jinhao.hu@pingcap.com>

### What problem does this PR solve?

Issue Number: Close #40598

Problem Summary:
as https://github.com/tikv/pd/pull/5861 described, will be merged after 5861 to make sure compatibility.

### Check List

Tests 

- [x] Unit test

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by nolouch** [comment]
@HuSharp plz update the client-go.

File: store/mockstore/unistore/pd.go [hunk 1]
```diff
@@ -44,26 +46,24 @@ func newPDClient(pd *us.MockPD) *pdClient {
...
 	}
-	return ret, nil
+	return ret, 0, nil
 }
 
-func (c *pdClient) StoreGlobalConfig(ctx context.Context, items []pd.GlobalConfigItem) error {
+func (c *pdClient) StoreGlobalConfig(ctx context.Context, configPath string, items []pd.GlobalConfigItem) error {
 	for _, item := range items {
-		c.globalConfig["/global/config/"+item.Name] = item.Value
+		c.globalConfig[configPath+item.Name] = item.Value
```
Comment (nolouch) [inline store/mockstore/unistore/pd.go:61]: better use `path.Join(configPath, item.Name)`.

**Comment by BornChanger** [comment]
is build job in ci fail expected ?

**Comment by HuSharp** [comment]
> /retest

I will update the client-go after the merge at https://github.com/tikv/client-go/pull/648.
But this error is not supposed to appear.

**Review by BornChanger — APPROVED** [review]

**Review by nolouch — APPROVED** [review]
lgtm

**Pull request merged**